src/utils/buildMetadata.ts

```diff
@@ -233,9 +425,33 @@ function buildProgramRuleVariables(sheets: Sheet[]) {
     });
 }
 
+// Add sortOrder to filteredMetadataItems, these items belong to the same 'group'.
+// The idea is to use metadataItems.filter(filterFunction) as filteredMetadataItems.
+function makeSortOrder(filteredMetadataItems: MetadataItem[]) {
+    filteredMetadataItems.forEach((item, index) => {
+        item.sortOrder = index;
+    });
+}
```

Comment:
The old method uses lodash methods, that's why its so different. 
I used a different approach because groupBy() didn't work for cases where we needed to group the elements by multiple fields (same program, programStage...). There is probably a way to make groupBy() work but i didn't find one and it was probably more complicated.

The makeSortOrder function receives all the metadata elements that belong to the same sortOrder 'pool' and assigns it by their position on the spreadsheet.

There is a difference, the old one starts with sortOrder 1, but DHIS2 starts with 0 when adding metadata so i did the same.